### PR TITLE
MOB-2721 - Integrate payment providers enpoint

### PR DESCRIPTION
**Pull request by jsm20**
## JIRA
[MOB-2721](https://karhoo.atlassian.net/browse/MOB-2721)

## DESCRIPTION
Integrate the new GET /v3/payments/providers for Adyen. 

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [x] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
N/a

### CODE REVIEW

File: KarhooSDK/Network/APIEndpoint.swift [hunk 1]
```diff
@@ -112,6 +113,8 @@ enum APIEndpoint {
             return "/oauth/v2/userinfo"
         case .authRefresh:
             return "/oauth/v2/token"
+        case .paymentProvider:
```
Comment (mohadian) [inline KarhooSDK/Network/APIEndpoint.swift:116]: minor comment, would it be better to be called paymentProvider**s**? as response is a list and also the API endpoint also has the `s`

File: KarhooSDK/Api/Response/PaymentProvider.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//
+//  Provider.swift
+//  KarhooSDK
+//
+//  Copyright © 2020 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+public struct PaymentProvider : KarhooCodableModel {
```
Comment (mohadian) [inline KarhooSDK/Api/Response/PaymentProvider.swift:10]: In Android, it is called [Provider](https://github.com/karhoo/karhoo-android-sdk/blob/develop/sdk/src/main/java/com/karhoo/sdk/api/model/Provider.kt), would be good to have consistency. I think it makes sense to call it PaymentProvider though. 

@jeremyauclert

File: KarhooSDK/Network/APIEndpoint.swift (hunk 1 above)
Comment (jsm20) [reply KarhooSDK/Network/APIEndpoint.swift:116]: Seems to be a single Provider rather than a list.....

Comment (mohadian) [reply KarhooSDK/Network/APIEndpoint.swift:116]: That's true at the moment. Apparently the PSP would be one until perhaps they introduce loyalty?

**Pull request merged**